### PR TITLE
fix(Control): `calcCornerCoords` angle + calculation

### PR DESCRIPTION
**Pull request by ShaMan123**
## Description

The angle passed to `calcCornerCoords` should be the angle from the viewport in order to respect nested objects. Since currently vpt rotation is not supported it is enough to get the angle in the scene plane but that will become a bug once vpt rotation is supported (e.g. #8767 )
Additionally the calculation was cumbersome and very hard to follow.
There were no tests for it, I added both unit and e2e tests, e2e to prove that my fix is correct.

## Changes

- angle arg
- calculation more concise using matrices instead of decomposed values
- type sizeX/Y touchSizeX/Y - no need for annoying null

## In Action

Actual is from master, expected is this PR

https://github.com/fabricjs/fabric.js/assets/34343793/d0b30d2c-6c32-4bea-b445-e22ac07a196d

### CODE REVIEW

**Review by ShaMan123 — COMMENTED** [review]
ready
If you want this to become more than one PR say so

**Review by ShaMan123 — COMMENTED** [review]
updated desc
validated the snapshot against master so all is good to go

**Comment by asturur** [comment]
@ShaMan123 #8767 is mentioned to corroborate any change.
please close it, it won't get merged in current status.
There is a super old PR for canvas rotation, if we get time to work on other things other the one we are doing now i will resume that pr and i ll have the pleasure to finish it.

**Comment by asturur** [comment]
The benchmark show that this approach other than terser is also a bit faster.

**Comment by asturur** [comment]
i need to look at the jest test, but i m too tired now, i ll do that first thing tomorrow

File: src/shapes/Object/FabricObject.spec.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    const spies = Object.values(controls).map((control) =>
+      jest.spyOn(control, 'calcCornerCoords')
+    );
+    group.setCoords();
+    expect(
+      Object.keys(controls).reduce((acc, key, index) => {
+        const { calls, results } = spies[index].mock;
+        return Object.assign(acc, { [key]: { calls, results } });
+      }, {})
+    ).toMatchSnapshot(roundSnapshotOptions);
```
[the remark this answers is not in the document]
Comment (asturur) [reply src/shapes/Object/FabricObject.spec.ts]: Is better to have tests that test one assertion, a snapshot covers a lot of things but doesn't tell you want wants to preserver. I changed the test to measure the angle of the corners

**Review by asturur — APPROVED** [review]
